src/pages/DatasetRegistration.js

```diff
@@ -244,6 +244,21 @@ class DatasetRegistration extends Component {
     });
   };
 
+  // same as handleChange, but adds call to validate dataset name and only affects state if a change has been made
+  handleDatasetNameChange = (e) => {
+    const value = e.target.value;
+    if (this.state.datasetData.datasetName !== value) {
+      this.validateDatasetName(value);
```

Comment:
This function is asynchronous, so the code (as is) will execute before ```validateDatasetName()``` resolves. This causes a side-effect where the error message shows briefly while the dataset name is being validated. Easy way to fix this is to make ```handleDatasetNameChange``` an async function and add an ```await``` keyword in front of ```validateDatasetName```. This way the body of the function will wait on the validation resolve before moving forward.